shell/platform/embedder/embedder_semantics_update.h

```diff
@@ -58,13 +58,23 @@ class EmbedderSemanticsUpdate2 {
   std::vector<FlutterSemanticsCustomAction2> actions_;
   std::vector<FlutterSemanticsCustomAction2*> action_pointers_;
 
+  std::vector<std::unique_ptr<std::vector<const FlutterStringAttribute*>>>
+      node_string_attributes_;
+  std::vector<std::unique_ptr<FlutterStringAttribute>> string_attributes_;
+  std::vector<std::unique_ptr<FlutterLocaleStringAttribute>> locale_attributes_;
+  std::unique_ptr<FlutterSpellOutStringAttribute> spell_out_attribute_;
+
   // Translates engine semantic nodes to embedder semantic nodes.
   void AddNode(const SemanticsNode& node);
 
   // Translates engine semantic custom actions to embedder semantic custom
   // actions.
   void AddAction(const CustomAccessibilityAction& action);
 
+  // Translates engine string attributes to embedder string attributes.
+  std::pair<size_t, const FlutterStringAttribute**> CreateStringAttributes(
+      const StringAttributes& attribute);
```

Comment:
Optional: Since this isn't public API, I'd be tempted to create a cheapo wrapper struct for the return value purely for readability, in order to avoid the whole std::pair/first,second, but don't feel particularly strongly about it. Then you could do something like:
```cc
return {
  .attributes = node_string_attributes_.back()->data(),
  .size = node_string_attributes_.back()->size()
};
```
or even just:
```cc
return {node_string_attributes_.back()->data(),
        node_string_attributes_.back()->size()};
```
similar to what we do now. As I say, I don't feel super strongly about it; mostly just helps readability at the site where this uses `.first` and `.second` now.

Reply:
Thanks for the suggestion, it does improve readability nicely. Added!